### PR TITLE
[script][combat-trainer] Apparent fix for load/aim loop

### PR DESCRIPTION
**Pull request by Raykyn55**
Inverted timer function, added incremental reduction in time-to-fire to break check_firing_time loops.

added @firing_check resets after each instance of game_state.loaded becoming false, in order that we not skip the initial condition on the first time check (which sets the disparity between now and next fire) when we start aiming after a fresh load.

added some debug lines to help with testing.

```
game_state.clear_aim_queue if Flags['ct-ranged-ready']
```
This line was leaving ct with nothing to do when full-aim message arrived, but the conditionals weren't passing the firing command(still not sure why) so i moved the flag opposite our timer. As long as EITHER the full aim message has arrived, OR the time has expired, we get a fire command, but until then, we keep cycling fillers.

Best way I've found to recreate the bug was commenting out aim_fillers entirely (4425 in the current version), do ranged weapons only, no offhand attacks, and only training abilities was combat-maneuvers.

### CODE REVIEW

**Comment by MahtraDR** [comment]
I've got this going locally, will report back.

File: combat-trainer.lic [hunk 1]
```diff
@@ -3225,7 +3234,10 @@ class AttackProcess
                   'shoot'
                 end
       shoot_aimed(command, game_state)
+      game_state.clear_aim_queue
       game_state.loaded = false
+      @firing_check = 0
+      echo("@firing_check reset normal fire: #{@firing_check}") if debug_mode_ct
```
Comment (MahtraDR) [inline combat-trainer.lic:3240]:
```suggestion
      echo("@firing_check reset normal fire: #{@firing_check}") if $debug_mode_ct
```

File: combat-trainer.lic [hunk 2]
```diff
@@ -3525,6 +3542,8 @@ class AttackProcess
           shoot_aimed('shoot', game_state)
         end
         game_state.loaded = false
+        @firing_check = 0
+        echo("@firing_check reset on maneuver fire: #{@firing_check}") if debug_mode_ct
```
Comment (MahtraDR) [inline combat-trainer.lic:3546]:
```suggestion
        echo("@firing_check reset on maneuver fire: #{@firing_check}") if $debug_mode_ct
```

**Comment by asechrest** [comment]
Thanks for the PR! I'll pull this down this evening and run it, though I've never seen the bug this tries to fix. So I'll just test for no adverse issues.

**Comment by Raykyn55** [comment]
> Thanks for the PR! I'll pull this down this evening and run it, though I've never seen the bug this tries to fix. So I'll just test for no adverse issues.

Most probably wouldnt, but with certain conditions, you can force it. Run blank aiming_trainables in the old one and youll see it more often, especially after powershot without dual load.

**Comment by XTyrsin** [comment]
Working fine in general and appears to have successfully resolved the load-aim-shoot bug.

**Comment by asechrest** [comment]
This has been through some testing and appears to fix the issue and not cause others. Will merge and monitor.

**Pull request merged**